src/fetch_cnpj_info.py

```diff
@@ -0,0 +1,114 @@
+                                     'logradouro',
+                                     'numero',
+                                     'cep',
+                                     'municipio',
+                                     'uf',
+                                     'abertura',
+                                     'natureza_juridica',
+                                     'fantasia',
+                                     'cnpj',
+                                     'ultima_atualizacao',
+                                     'status',
+                                     'complemento',
+                                     'email',
+                                     'efr',
+                                     'motivo_situacao',
+                                     'situacao_especial',
+                                     'data_situacao_especial'])
+
+def read_csv(name):
+    return pd.read_csv('data/2016-08-08-%s.xz' % name,
+                       parse_dates=[16],
+                       dtype={'document_id': np.str,
+                              'congressperson_id': np.str,
+                              'congressperson_document': np.str,
+                              'term_id': np.str,
+                              'cnpj_cpf': np.str,
+                              'reimbursement_number': np.str})
+
+def remaining_cnpjs(info_dataset):
+    datasets = [read_csv(name)
+                for name in ['current-year', 'last-year', 'previous-years']]
+    dataset = pd.concat(datasets)
+    del(datasets)
+    is_cnpj = dataset['cnpj_cpf'].str.len() == 14.
+    cnpj_list = set(dataset.loc[is_cnpj, 'cnpj_cpf'])
+    already_fetched = set(info_dataset['cnpj'].str.replace(r'[./-]', ''))
+    return list(cnpj_list - already_fetched)
+
+def fetch_cnpj_info(cnpj, timeout=5):
+    url = 'http://receitaws.com.br/v1/cnpj/%s' % cnpj
+    print('Fetching %s' % cnpj)
+    json_contents = urlopen(url, timeout=timeout).read().decode('utf-8')
+    return json.loads(json_contents)
+
+def write_cnpj_info(cnpj, cnpj_info):
+    print('Writing %s' % cnpj)
+    with open('%s/%s.pkl' % (TEMP_PATH, cnpj), 'wb') as f:
+        pickle.dump(cnpj_info, f, pickle.HIGHEST_PROTOCOL)
+
+def read_cnpj_info(cnpj_filename):
+    with open('%s/%s' % (TEMP_PATH, cnpj_filename), 'rb') as f:
+        return pickle.load(f)
+
+def import_cnpj_infos(info_dataset):
+    cnpjs_to_import = [filename
+                       for filename in os.listdir(TEMP_PATH)
+                       if filename.endswith('.pkl')]
+    for filename in cnpjs_to_import:
+        print('Importing %s' % filename)
+        attributes = read_cnpj_info(filename)
+        info_dataset = info_dataset.append(attributes, ignore_index=True)
+    info_dataset.to_csv(INFO_DATASET_PATH,
+                        compression='xz',
+                        encoding='utf-8',
+                        index=False)
+
+
+
+if not os.path.exists(TEMP_PATH):
+    os.makedirs(TEMP_PATH)
```

Comment:
If this is really a temporary file, we might use [`TemporaryDirectory`] (https://docs.python.org/3.5/library/tempfile.html#tempfile.TemporaryDirectory) — I can code that, pairing or not if you feel like we should use it. We can also have [an object in memory that works like a file](https://docs.python.org/3.5/library/io.html) (not sure about if this is a proper approach though… I mean, if this gets too big we'll be requiring tons of RAM).